operator/pkg/ciliumidentity/reconciler.go

```diff
@@ -163,6 +160,11 @@ func (r *reconciler) reconcileCID(cidResourceKey resource.Key) error {
 			return r.createCID(cidName, cidKey)
 		} else {
 			r.desiredCIDState.Remove(cidName)
+			err := r.makeIDAvailable(cidName)
+			r.logger.Warn("Failed to return CID to pool",
```

Comment:
We should log only in case of err != nil

Reply:
Fixed